runtime/shared/shrclssup.c

```diff
@@ -141,6 +141,23 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void* reserved)
 					runtimeFlags |= J9SHR_RUNTIMEFLAG_DISABLE_BCI;
 				}
 
+				/* Check for -XX:+PortableSharedCache and -XX:-PortableSharedCache; whichever comes later wins.
+				 * These options should be checked before parseArgs() to allow -Xshareclasses:portable to override this option.
+				 * 
+				 * Note: Please also change the function checkArgsConsumed() in runtime/vm/jvminit.c when adding new options,
+				 * in order to quietly consume the options if it is used without -Xshareclasses
+				 */
+				argIndex1 = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXPORTABLESHAREDCACHE, NULL);
+				argIndex2 = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOPORTABLESHAREDCACHE, NULL);
```

Comment:
Is the JIT code that checks sharedCachePortable inside `#if defined(J9X86)` ? If not, you may want to put these 2 FIND_AND_CONSUME_ARG calls inside `#if defined(J9X86)`

Reply:
@hangshao0 Actually I just found out (local testing) that `J9X86` is not defined here. Is there another MACRO we can use?